src/Project.hs

```diff
@@ -510,3 +553,44 @@ isIntegratedOrSpeculativelyConflicted pr =
   (Integrated _ _)                            -> True
   (Conflicted base _) | base /= baseBranch pr -> True
   _                                           -> False
+
+latestStatus :: OutstandingChecks -> BuildStatus
+latestStatus (AnyCheck status) = status
+latestStatus (SpecificChecks statusChecks) =
+  let statuses = map snd $ Map.toList statusChecks
+  in if null statuses then BuildPending else maximum statuses
```

Comment:
It's not really a simple order relationship, right? Because now, as soon as one check succeeds, we'd see the whole thing as succeeded, even if there are still outstanding checks.

Reply:
Good one, I was hoping to simplify the logic a bit using `Ord`, but it indeed doesn't cover all bases.